jina/types/request/data.py

```diff
@@ -127,15 +128,25 @@ def __init__(
                 f'fail to construct a {self.__class__} object from {request}'
             ) from ex
 
-        self._pb_body_wo_data = None
-
     @property
     def is_decompressed(self) -> bool:
+        """
+        Checks if the underlying proto object was already deserialized into a :class:`jina.proto.jina_pb2.DataRequestProto` or
+        :class:`jina.proto.jina_pb2.DataRequestProtoWoData`
+           :return: True if the proto was deserialized before
+        """
+        return type(self._pb_body) in [
```

Comment:
why was last logic not valid?

Reply:
It rellied on the fact that buffer is None if _pb_body. This one is more precise and clear with the new change